### PR TITLE
extend documentation to show how to use only native CMake commands

### PR DESCRIPTION
**Pull request by davydden**
based on @tamiko comments here https://github.com/LLNL/spack/pull/816

fixes https://github.com/dealii/dealii/issues/2541

### CODE REVIEW

**Comment by bangerth** [comment]
I think @tamiko has to review this one :-)

**Comment by davydden** [comment]
that would be my guess as well ;-)

File: doc/users/cmakelists.html [hunk 1]
```diff
@@ -529,6 +530,42 @@
...
+user project using only native CMake commands, that is avoid usage of
+<code>DEAL_II_INITIALIZE_CACHED_VARIABLES</code> and <code>DEAL_II_SETUP_TARGET</code> macros.
+This can be achieved, for example, by the following code
+<pre class="cmake">
+SET(TARGET "mycode")
+
+# Declare all source files the target consists of.
+SET(TARGET_SRC
+  ${TARGET}.cc
+  )
```
Comment (tamiko) [inline doc/users/cmakelists.html]: Please remove the variables ```TARGET``` and ```TARGET_SRC``` and put it directly into the statements.

File: doc/users/cmakelists.html [hunk 2]
```diff
@@ -529,6 +530,42 @@
...
+
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8.8)
```
Comment (tamiko) [inline doc/users/cmakelists.html]: Please require version 3.0.0

File: doc/users/cmakelists.html [hunk 3]
```diff
@@ -529,6 +530,42 @@
...
+
+# Find and include deal.IIConfig.cmake
+FIND_PACKAGE(deal.II 8.5.0 QUIET
+  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
+  )
```
Comment (tamiko) [inline doc/users/cmakelists.html]: Please change ```QUIET``` to ```REQUIRED```.

File: doc/users/cmakelists.html [hunk 4]
```diff
@@ -529,6 +530,42 @@
...
+
+PROJECT(${TARGET})
```
Comment (tamiko) [inline doc/users/cmakelists.html]: ```PROJECT(example)```

File: doc/users/cmakelists.html [hunk 5]
```diff
@@ -529,6 +530,42 @@
...
+
+# Include deal.IITargets.cmake
+INCLUDE(${DEAL_II_TARGET_CONFIG})
+
+ADD_EXECUTABLE(myexec ${TARGET_SRC})
+TARGET_LINK_LIBRARIES(myexec ${DEAL_II_LIBRARIES})
+INCLUDE_DIRECTORIES(${DEAL_II_INCLUDE_DIRS})
+SET(CMAKE_CXX_FLAGS ${DEAL_II_CXX_FLAGS})
+SET(CMAKE_CXX_COMPILER ${DEAL_II_CXX_COMPILER})
```
Comment (tamiko) [inline doc/users/cmakelists.html]: Replace by
```
ADD_EXECUTABLE(example example.src)
TARGET_LINK_LIBRARIES(example ${DEAL_II_TARGET})
```

All the other statements are unnecessary once #2543 is merged.

**Comment by tamiko** [comment]
In summary:

```
CMAKE_MINIMUM_REQUIRED(VERSION 3.0.0)
# Find and include deal.IIConfig.cmake
FIND_PACKAGE(deal.II 8.5.0 REQUIRED
  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
  )
# Include deal.IITargets.cmake
INCLUDE(${DEAL_II_TARGET_CONFIG})

ADD_EXECUTABLE(example example.src)
TARGET_LINK_LIBRARIES(example ${DEAL_II_TARGET})
```

**Comment by davydden** [comment]
@tamiko: i update the PR.

**Pull request merged**